tools/llvm-bpf: LLVM 18.x fails to compile with GCC 15+

### Describe the bug

There's a fix for this in [LLVM 20.x](https://github.com/llvm/llvm-project/commit/7e44305041d96b064c197216b931ae3917a34ac1), backported to [19.x](https://github.com/llvm/llvm-project/pull/101771) but not 18.x.

As a shortcut, adding the relevant [patch](https://github.com/llvm/llvm-project/commit/7e44305041d96b064c197216b931ae3917a34ac1.patch) to `tools/llvm-bpf/patches` works. Not sure what the difficulty/implications of an LLVM upgrade are, but that might be a good idea too.

### OpenWrt version

ffde9a9fe9adcb028ab1b39ae0728bfffe4ec304

### OpenWrt release

ffde9a9fe9adcb028ab1b39ae0728bfffe4ec304

### OpenWrt target/subtarget

N/A

### Device

N/A

### Image kind

Self-built image

### Steps to reproduce

```
git clone git@github.com:openwrt/openwrt.git
cd openwrt && git checkout ffde9a9fe9adcb028ab1b39ae0728bfffe4ec304
scripts/feeds update -a
scripts/feeds install -a

# Fast compile dependencies, adjust to system threads
make tools/llvm-bpf/compile -j8
# Debug output
make tools/llvm-bpf/compile -j1 V=s
```

### Actual behaviour

```
/home/htung/dev/openwrt/staging_dir/host/bin/g++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/build/lib/Support -I/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/lib/Support -I/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/build/include -I/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/include -I/home/htung/dev/openwrt/staging_dir/host/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/Allocator.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/Allocator.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/Allocator.cpp.o -c /home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/lib/Support/Allocator.cpp
In file included from /home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/include/llvm/Support/Allocator.h:20,
                 from /home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/lib/Support/Allocator.cpp:13:
/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/include/llvm/ADT/SmallVector.h:119:62: error: 'uint64_t' was not declared in this scope
  119 |     std::conditional_t<sizeof(T) < 4 && sizeof(void *) >= 8, uint64_t,
      |                                                              ^~~~~~~~
/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/include/llvm/ADT/SmallVector.h:29:1: note: 'uint64_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
   28 | #include <memory>
  +++ |+#include <cstdint>
   29 | #include <new>
/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/include/llvm/ADT/SmallVector.h:120:24: error: 'uint32_t' was not declared in this scope
  120 |                        uint32_t>;
      |                        ^~~~~~~~
/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/include/llvm/ADT/SmallVector.h:120:24: note: 'uint32_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/include/llvm/ADT/SmallVector.h:120:32: error: template argument 2 is invalid
  120 |                        uint32_t>;
      |                                ^
/home/htung/dev/openwrt/build_dir/host/llvm-project-18.1.7.src/llvm/include/llvm/ADT/SmallVector.h:120:32: error: template argument 3 is invalid
... etc etc ...
```


### Expected behaviour

llvm-bpf compiles successfully.

### Additional info

```
$ g++ --version
g++ (GCC) 15.1.1 20250425
Copyright (C) 2025 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

More context: https://gcc.gnu.org/gcc-15/porting_to.html#header-dep-changes

### Diffconfig

```text
$ ./scripts/diffconfig.sh 
CONFIG_TARGET_mediatek=y
CONFIG_TARGET_mediatek_filogic=y
CONFIG_TARGET_mediatek_filogic_DEVICE_openwrt_one=y
```

### Terms

- [x] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/tools/llvm-bpf/patches/100-gcc15.patch
+++ b/tools/llvm-bpf/patches/100-gcc15.patch
@@ -1,0 +1,23 @@
+From 7e44305041d96b064c197216b931ae3917a34ac1 Mon Sep 17 00:00:00 2001
+From: Sam James <sam@gentoo.org>
+Date: Fri, 2 Aug 2024 23:07:21 +0100
+Subject: [PATCH] [ADT] Add `<cstdint>` to SmallVector (#101761)
+
+SmallVector uses `uint32_t`, `uint64_t` without including `<cstdint>`
+which fails to build w/ GCC 15 after a change in libstdc++ [0]
+
+[0] https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=3a817a4a5a6d94da9127af3be9f84a74e3076ee2
+---
+ llvm/include/llvm/ADT/SmallVector.h | 1 +
+ 1 file changed, 1 insertion(+)
+
+--- a/llvm/include/llvm/ADT/SmallVector.h
++++ b/llvm/include/llvm/ADT/SmallVector.h
+@@ -19,6 +19,7 @@
+ #include <algorithm>
+ #include <cassert>
+ #include <cstddef>
++#include <cstdint>
+ #include <cstdlib>
+ #include <cstring>
+ #include <functional>

